### PR TITLE
Update draft-ietf-dnsop-serve-stale.md

### PR DESCRIPTION
**Pull request by puneetsood**
Minor edits for clarity. Some comments in the implementation considerations section.

### CODE REVIEW

File: draft-ietf-dnsop-serve-stale.md [hunk 1]
```diff
@@ -67,7 +67,9 @@ data to avoid outages when authoritative nameservers cannot be reached
 to refresh expired data.  It updates the definition of TTL from
 {{!RFC1034}} and {{!RFC1035}} to make it clear that data can be kept
 in the cache beyond the TTL expiry and used for responses when a
-refreshed answer is not readily available.
+refreshed answer is not readily available. One of the motivations for
+serve-stale is to make the DNS more resilient to DOS attacks (and
+thereby make them less attractive as an attack vector).
```
Comment (wkumari) [inline draft-ietf-dnsop-serve-stale.md:73]: Nice.

File: draft-ietf-dnsop-serve-stale.md [hunk 2]
```diff
@@ -297,25 +299,24 @@ document.
 
 ## Implementation considerations
 
-This document mainly describes the method / concept behind serving stale data, and intentionally does not provide a formal algorithm. The concept is not overly complex, and we believe that it would be hubris to believe that we know better than resolver authors how exactly to implement the concept in their code-base. The processing of serve-stale is a local operation, and consitent variables between implementations (or instances) is not needed for interoperability. 
+This document mainly describes the method / concept behind serving stale data, and intentionally does not provide a formal algorithm. The concept is not overly complex, and the details are best left to resolver authors to implement the concept in their code-base. The processing of serve-stale is a local operation, and consitent variables between implementations (or instances) are not needed for interoperability.
```
Comment (wkumari) [inline draft-ietf-dnsop-serve-stale.md:303]: ... fine. I liked my hubris comment, but this is probably more readable... <grump>

File: draft-ietf-dnsop-serve-stale.md [hunk 3]
```diff
@@ -297,25 +299,24 @@ document.
...
 
 The "client response timer" is another variable which deserves consideration. If this value is too short, there exists the risk that stale answers may be used even when the authoritative server is actually reachable but slow; this may result in suboptimal answers being returned. Conversely, waiting too long will negatively impact the user response. 
 
-Many resolvers implement prefetching of answers before the TTL has expired. If this has been attempted recently, and the authorative servers were determined to be not reachable at this time, the check can be skipped - the authors recommend that this value be the same as the check interval (see below). 
+Many resolvers implement prefetching of answers before the TTL has expired. If this has been attempted recently, and the authorative servers were determined to be not reachable at this time, the check can be skipped - the authors recommend that this value be the same as the check interval (see below).
```
Comment (wkumari) [inline draft-ietf-dnsop-serve-stale.md:310]: Nice.

File: draft-ietf-dnsop-serve-stale.md [hunk 4]
```diff
@@ -297,25 +299,24 @@ document.
...
+Another variable is how often to re-check if the authoritative server is available once it has been determined to be unreachable. If this variable is set too large, stale answers may continue to be returned even after the authoritative server is reachable. If the serve-stale technique becomes widely deployed, and this variable is too small, authoritative servers may be hit with a significant amount of traffic when they become reachable again. For this reason, the authors strongly recommend that this value not be set lower than 30 seconds.
```
Comment (wkumari) [inline draft-ietf-dnsop-serve-stale.md:312]: Ok. I still think it is worth having something saying that the very fact that serve-stale exists should decrease the occurrence of DoS attacks, but I'm not sure where to put it...

**Review by wkumari — APPROVED** [review]

**Comment by wkumari** [comment]
I merged in the comments.

File: draft-ietf-dnsop-serve-stale.md (hunk 4 above)
Comment (puneetsood) [reply draft-ietf-dnsop-serve-stale.md:312]: I moved that sentence to the top of the draft. In this paragraph it felt out of place in terms of the level of detail.

**Pull request merged**